### DISCUSSION TITLE
Which part of SWC should be documented

### DISCUSSION THREAD

**Discussion by kdy1** [swc-project/swc#5776] [General]
I'd love tho hear about opinions about documentation

**Comment by jjnesbitt** [comment]
I'm not sure if this is a focus, but I'm currently using SWC to transform JS code into different JS code, not compiling it directly. I think documentation on what the "correct" way to do that is would be very helpful to a newcomer, as I've found myself having to rely on the source code as my primary way to figure this out. As a result I'm probably doing some things wrong still (for example comments still elude me). The plugin [docs](https://swc.rs/docs/plugin/ecmascript/getting-started) help, but there's nothing mentioning emitters for example.

**Reply by kdy1** [reply]
There are lots of examples in swc repository. SWC simply follows the convention of cargo instead of trying something different.

https://github.com/swc-project/swc/blob/f739976bdd370abda9d5315a2f0a99411a7620d3/crates/swc_ecma_codegen/examples/gen.rs

**Reply by jjnesbitt** [reply]
Are there any examples of generating _new_ comments, or preserving comments from the source into newly emitted code? The main two issues I have right now are:

* I want to preserve existing comments as well as add new ones.
* The generated code has very little whitespace. I'd like either preserve the input whitespace if possible, or add manually in whitespace to the newly emitted code.

**Comment by RiESAEX** [comment]
resolver and syntaxContext, I still don't understand how they work.

**Reply by kdy1** [reply]
Is https://rustdoc.swc.rs/swc_ecma_transforms_base/fn.resolver.html not enough?
